pyigm/field/igmfield.py

```diff
@@ -0,0 +1,236 @@
+""" Class for correlating galaxies and IGM within a given field on the sky
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+import numpy as np
+import warnings
+import pdb
+
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+from astropy import constants as const
+
+#from xastropy.xutils import xdebug as xdb
+
+
+class IgmGalaxyField(object):
+    """ Class for a field associating galaxies to the IGM/CGM
+
+    Parameters
+    ----------
+    radec : tuple or SkyCoord
+        (RA,DEC) in deg or astropy.coordinate
+    name : str, optional
+    """
+
+    # Initialize 
+    def __init__(self, radec, name='', cosmo=None, verbose=False):
+        # coord
+        if isinstance(radec, (tuple)):
+            self.coord = SkyCoord(ra=radec[0], dec=radec[1])
+        elif isinstance(radec, SkyCoord):
+            self.coord = radec
+
+        # Field
+        self.name = name
+
+        # Cosmology
+        if cosmo is None:
+            from astropy.cosmology import WMAP9 as cosmo
+            if verbose is True:
+                print('IgmGalxyField: Using WMAP9 cosmology')
+        self.cosmo = cosmo
+
+        # Init
+        self.igm = None
+        self.targets = None
+        self.galaxies = None
+        self.observing = None
+        self.selection = None
+
+    def calc_rhoimpact(self, obj, los_coord=None):
```

Comment:
Leave as an optional parameter whether the calculation is in physical or co-moving coordinates. I would make the default to be co-moving though.